diff -up scrcpy-1.21/app/src/decoder.h.omv~ scrcpy-1.21/app/src/decoder.h
--- scrcpy-1.21/app/src/decoder.h.omv~	2022-01-16 01:39:33.531714732 +0100
+++ scrcpy-1.21/app/src/decoder.h	2022-01-16 01:39:36.598726354 +0100
@@ -16,7 +16,7 @@ struct decoder {
     struct sc_frame_sink *sinks[DECODER_MAX_SINKS];
     unsigned sink_count;
 
-    AVCodecContext *codec_ctx;
+    struct AVCodecContext *codec_ctx;
     AVFrame *frame;
 };
 
diff -up scrcpy-1.21/app/src/icon.c.omv~ scrcpy-1.21/app/src/icon.c
--- scrcpy-1.21/app/src/icon.c.omv~	2022-01-16 01:39:44.139754948 +0100
+++ scrcpy-1.21/app/src/icon.c	2022-01-16 01:39:47.033765927 +0100
@@ -91,7 +91,7 @@ decode_image(const char *path) {
         goto close_input;
     }
 
-    AVCodecContext *codec_ctx = avcodec_alloc_context3(codec);
+    struct AVCodecContext *codec_ctx = avcodec_alloc_context3(codec);
     if (!codec_ctx) {
         LOG_OOM();
         goto close_input;
diff -up scrcpy-1.21/app/src/stream.c.omv~ scrcpy-1.21/app/src/stream.c
--- scrcpy-1.21/app/src/stream.c.omv~	2022-01-16 01:43:02.647515101 +0100
+++ scrcpy-1.21/app/src/stream.c	2022-01-16 01:43:25.754604393 +0100
@@ -2,6 +2,7 @@
 
 #include <assert.h>
 #include <libavformat/avformat.h>
+#include <libavcodec/avcodec.h>
 #include <libavutil/time.h>
 #include <unistd.h>
 
@@ -192,7 +193,7 @@ static int
 run_stream(void *data) {
     struct stream *stream = data;
 
-    AVCodec *codec = avcodec_find_decoder(AV_CODEC_ID_H264);
+    const AVCodec *codec = avcodec_find_decoder(AV_CODEC_ID_H264);
     if (!codec) {
         LOGE("H.264 decoder not found");
         goto end;
diff -up scrcpy-1.21/app/src/stream.h.omv~ scrcpy-1.21/app/src/stream.h
--- scrcpy-1.21/app/src/stream.h.omv~	2022-01-16 01:39:54.835795543 +0100
+++ scrcpy-1.21/app/src/stream.h	2022-01-16 01:40:00.088815498 +0100
@@ -20,8 +20,8 @@ struct stream {
     struct sc_packet_sink *sinks[STREAM_MAX_SINKS];
     unsigned sink_count;
 
-    AVCodecContext *codec_ctx;
-    AVCodecParserContext *parser;
+    struct AVCodecContext *codec_ctx;
+    struct AVCodecParserContext *parser;
     // successive packets may need to be concatenated, until a non-config
     // packet is available
     AVPacket *pending;
diff -up scrcpy-1.21/app/src/v4l2_sink.c.omv~ scrcpy-1.21/app/src/v4l2_sink.c
--- scrcpy-1.21/app/src/v4l2_sink.c.omv~	2022-01-16 01:41:11.334087178 +0100
+++ scrcpy-1.21/app/src/v4l2_sink.c	2022-01-16 01:41:19.546118611 +0100
@@ -5,6 +5,8 @@
 #include "util/log.h"
 #include "util/str.h"
 
+#include <libavcodec/avcodec.h>
+
 /** Downcast frame_sink to sc_v4l2_sink */
 #define DOWNCAST(SINK) container_of(SINK, struct sc_v4l2_sink, frame_sink)
 
diff -up scrcpy-1.21/app/src/v4l2_sink.h.omv~ scrcpy-1.21/app/src/v4l2_sink.h
--- scrcpy-1.21/app/src/v4l2_sink.h.omv~	2022-01-16 01:40:07.272842805 +0100
+++ scrcpy-1.21/app/src/v4l2_sink.h	2022-01-16 01:40:10.002853188 +0100
@@ -15,7 +15,7 @@ struct sc_v4l2_sink {
 
     struct sc_video_buffer vb;
     AVFormatContext *format_ctx;
-    AVCodecContext *encoder_ctx;
+    struct AVCodecContext *encoder_ctx;
 
     char *device_name;
     struct sc_size frame_size;
